aws/rust-runtime/aws-sigv4/src/lib.rs

```diff
@@ -59,14 +56,17 @@ impl<'a, S> SigningParams<'a, S> {
     pub fn service_name(&self) -> &str {
         self.service_name
     }
+
+    /// If the identity in params contains AWS credentials, return them. Otherwise, return `None`.
+    pub fn credentials(&self) -> Option<&Credentials> {
+        self.identity.data::<Credentials>()
+    }
 }
 
 impl<'a, S: fmt::Debug> fmt::Debug for SigningParams<'a, S> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_struct("SigningParams")
-            .field("access_key", &"** redacted **")
-            .field("secret_key", &"** redacted **")
-            .field("security_token", &"** redacted **")
+            .field("identity", &"** redacted **")
```

Comment:
might as well delegate to `Identity` and trust its redaction? I guess this is safer but it might be helpful to, e.g. see the expiration time. And if we can't trust the debug impl on identity we have bigger issues

Reply:
hmm and then we don't need to impl debug anymore at all actually

Reply:
The `Identity` debug impl can't really redact anything itself, but it delegates to the inner `Debug`, which for `Credentials`, is already redacted. It should be safe to use its `Debug` implementation directly.